oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/endpoint/AbstractWebClientReactiveOAuth2AccessTokenResponseClient.java

```diff
@@ -97,7 +102,18 @@ private void populateTokenRequestHeaders(T grantRequest, HttpHeaders headers) {
 		headers.setAccept(Collections.singletonList(MediaType.APPLICATION_JSON));
 		if (ClientAuthenticationMethod.CLIENT_SECRET_BASIC.equals(clientRegistration.getClientAuthenticationMethod())
 				|| ClientAuthenticationMethod.BASIC.equals(clientRegistration.getClientAuthenticationMethod())) {
-			headers.setBasicAuth(clientRegistration.getClientId(), clientRegistration.getClientSecret());
+			try {
+				String clientId = this.encodeClientCredentials
+						? URLEncoder.encode(clientRegistration.getClientId(), StandardCharsets.UTF_8.displayName())
```

Comment:
I believe `StandardCharsets.UTF_8.displayName()` should be `StandardCharsets.UTF_8.name()`

Reply:
I went with `.toString()` to be consistent with elsewhere.